source/common/config/utility.cc

```diff
@@ -228,5 +228,18 @@ Grpc::AsyncClientFactoryPtr Utility::factoryForGrpcApiConfigSource(
   return async_client_manager.factoryForGrpcService(grpc_service, scope, false);
 }
 
+std::string Utility::normalizePath(absl::string_view path) {
+  std::vector<absl::string_view> segments = StringUtil::splitToken(path, "/", false);
+
+  // Removes dot and double-dot segments https://tools.ietf.org/html/rfc3986#section-5.2.4.
```

Comment:
@zuercher I think I will remove that partial support for now. 

A side question/clarification,

> It's unfortunate that we can't distinguish between prefix_rewrite = "" and unset because the real problem is that you can't rewrite a prefix to the empty string.

Do you think it is better to allow users to set prefix_rewrite as "" rather than "/"?